### PR TITLE
fix/bug grouping findings can report wrong vulnerabilities

### PR DESCRIPTION
**Pull request by oussamaessaji**
Do not merge until https://github.com/Ostorlab/agent_osv/pull/46 gets merged

### CODE REVIEW

File: tests/osv_agent_test.py [hunk 1]
```diff
@@ -358,3 +358,19 @@ def testAgentOSV_whenUnicodeDecodeError_shouldNotCrash(
     test_agent.process(scan_message_file)
 
     assert len(agent_mock) == 0
+
+
+def testAgentOSV_whenNoFindingsFromTheApi_returnsNoVulnz(
+    test_agent: osv_agent.OSVAgent,
+    agent_mock: list[message.Message],
+    agent_persist_mock: dict[str | bytes, str | bytes],
+    osv_api_output_risk_invalid: dict[str, Any],
```
Comment (deadly-panda) [inline tests/osv_agent_test.py:367]: Where is this used ?

File: agent/api_manager/osv_service_api.py [hunk 2]
```diff
@@ -57,8 +57,9 @@ def query_osv_api(
 
     response = requests.post(OSV_ENDPOINT, json=data)
 
-    if response.status_code == 200:
-        resp: dict[str, Any] = response.json()
-        return resp
-
-    return None
+    if response.status_code != 200:
+        return None
```
Comment (3asm) [inline agent/api_manager/osv_service_api.py:61]: This empty check should not be here.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/osv_agent_test.py [hunk 3]
```diff
@@ -358,3 +358,19 @@ def testAgentOSV_whenUnicodeDecodeError_shouldNotCrash(
...
+) -> None:
+    """Ensure that the agent does not detect vulnerabilities if the api returns no findings."""
+    selector = "v3.fingerprint.file.library"
+    msg_data = {"library_name": "jquery", "library_version": "3.6.0"}
+    msg = message.Message.from_data(selector, data=msg_data)
```
Comment (deadly-panda) [inline tests/osv_agent_test.py:372]: I would just mock the response.. If a cve is associated with this fingerprint the unit test will fail.

File: agent/osv_output_handler.py [hunk 4]
```diff
@@ -191,6 +191,8 @@ def parse_vulnerabilities_osv_api(
     references: List[dict[str, Any]] = []
     description = ""
     highest_risk_vuln_info: dict[str, str] = {}
+    if len(vulnerabilities) == 0:
+        return []
```
Comment (ErebusZ) [inline agent/osv_output_handler.py:195]: you return empty list here but the check you did is for None value

**Review by 3asm — APPROVED** [review]

**Pull request merged**